src/Validation/Constraint/HasClaimWithValue.php

```diff
@@ -0,0 +1,37 @@
+<?php
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Validation\Constraint;
+
+use Lcobucci\JWT\Token;
+use Lcobucci\JWT\UnencryptedToken;
+use Lcobucci\JWT\Validation\Constraint;
+use Lcobucci\JWT\Validation\ConstraintViolation;
+
+final class HasClaimWithValue implements Constraint
+{
+    private string $claimId;
+
+    private string $claimValue;
+
+    public function __construct(string $claimId, string $claimValue)
```

Comment:
The value can be anything, not only a string.

```suggestion
    private mixed $claimValue;

    public function __construct(string $claimId, mixed $claimValue)
```

Reply:
Only thing is there is no `mixed` type in PHP below 8.0

Reply:
😭  true that, something to be done soon

Reply:
Hmmm, should I change this to HasClaimWithStringValue.php, with the expectation to move to a more generic mixed when the library requires >php8.0 ?

Reply:
To handle the type difference in types you commented on? I can go with whatever, but atm this validation won't be able to do a claim with a different type like a boolean:

`isAdmin: true`

What do you want me to do here?

Reply:
> atm this validation won't be able to do a claim with a different type like a boolean:

Using `mixed` allows that and I believe that's the value of having a single implementation.

> What do you want me to do here?

I don't see value in having a string only constraint.
Not declaring a type already defines it as `mixed` and soon we'll drop v7.x support (according to [packagist](https://packagist.org/packages/lcobucci/jwt/php-stats#4) we have about 60% of adoption) we put the type declaration.

Reply:
That's a fair point

Reply:
Just to expand on the reason for a closure... the initial motivation to introduce this is related to hash comparison and doing that operation with `===` opens a security issue (timing attacks) - that's why we use `hash_equals()`. 

I'd really love to avoid having two different constraints in the lib for that purpose...